### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by KDlamini**
In this feature, we implement the following.

Add unit tests using Rspec to classes:
- `Book`
- `Person`
- `Student`
- `Teacher`
- `Classroom`
- `Corrector`
- `Rental`

Following the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: spec/corrector_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  context 'It should add a classroom' do
+    corrector = Corrector.new
+
+    it 'corrector should be an instance of class Corrector' do
+      expect(corrector).to be_an_instance_of(Corrector)
+    end
+
+    it 'should correct name with first letter in upper case' do
+      expect(corrector.correct_name('hamid')).to eq 'Hamid'
+    end
```
Comment (mikethreels) [inline spec/corrector_spec.rb:13]: - [ ] the `Corrector` is also responsible for making sure the name is not longer than 10 characters, please write a test for that as well.

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      expect(person.validate_name).not_to eq('hamid')
+    end
+
+    it 'should check age of Person' do
+      expect(person.age).to eq(23)
+    end
+
+    it 'should return true if he can use service' do
+      expect(person.can_use_services?).to eq true
+    end
```
Comment (mikethreels) [inline spec/person_spec.rb]: - [ ] please be as thorough as possible when writing tests, check if the code is also working when `can_use_services? = false`

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @KDlamini & @hamid-murambiwa, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**